src/dipdup/dipdup.py

```diff
@@ -248,7 +247,7 @@ async def _subscribe_to_datasource_events(self) -> None:
             datasource.call_on_events(self._on_events)
             datasource.call_on_rollback(self._on_rollback)
 
-    async def _on_head(self, datasource: IndexDatasource, head: TzktHeadBlockData) -> None:
+    async def _on_head(self, datasource: TzktDatasource, head: TzktHeadBlockData) -> None:
```

Comment:
Prob need to refactor this and the following proxy handlers to replace them with a generic one if possible. Will need to introduce evm specific otherwise, right?

Reply:
I'd postpone this to Phase III. I have no idea what the realtime part looks like in archives for now.